src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -1486,12 +1501,25 @@ private void handleClientExceptions(
    *     thrown by the WSM client or the retries
    */
   private <T> T handleClientExceptions(
-      HttpUtils.SupplierWithCheckedException<T, ApiException> makeRequest, String errorMsg) {
+      HttpUtils.SupplierWithCheckedException<T, ApiException> makeRequest,
+      String errorMsg,
+      boolean isDataCollectionWorkspace) {
     try {
       return makeRequest.makeRequest();
     } catch (ApiException | InterruptedException ex) {
       // if this is a WSM client exception, check for a message in the response body
       if (ex instanceof ApiException) {
+
+        // If requester doesn't have access to data collection workspace, normally
+        // `terra resource describe` would return:
+        //     [ERROR] Error fetching workspace: User XXX is not authorized to read resource YYY of
+        //     type workspace
+        // We generally hide the fact that data collections are workspaces. Show a different error
+        // that doesn't have "workspace".
+        if (isDataCollectionWorkspace) {
```

Comment:
I changed from `read access` to `access`. Hmm, I think not having read access will be the most common scenario. So for simplicity, I won't check status code. It's tricky because this could be any endpoint.